### PR TITLE
add blackd recipe output

### PR DESCRIPTION
**Pull request by carlodri**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
> Why is this a second output and not simply an added entrypoint?

because requirements are different/optional, and I didn't want to pollute requirements. Are you against this? I'm happy with your solution too.

**Comment by xhochy** [comment]
> > Why is this a second output and not simply an added entrypoint?
> 
> because requirements are different/optional, and I didn't want to pollute requirements. Are you against this? I'm happy with your solution too.

Ah that makes sense. You probably should though only build a meta-package for `blackd` that installs the requirements, otherwise `black` and `blackd` would conflict as they override each other.

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
ooh, thanks I didn't know about meta-packages, can you give some hints? Maybe it's in the docs, let me check..

**Comment by xhochy** [comment]
meta-packages are no special packages, they are just packages without actual content. Have a look at how the split between `matplotlib` and `matplotlib-base` is done:

https://github.com/conda-forge/matplotlib-feedstock/blob/f37cab6231e746cc0781070552defdeaf12475dd/recipe/meta.yaml#L71-L79 

`matplotlib` is just a meta-package that includes the Qt dependency.

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@xhochy what do you think about this?

**Comment by carlodri** [comment]
I'm not sure about this pin_subpackage stuff...

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@xhochy I guess I'm stuck here...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,33 +10,48 @@ source:
...
+        - appdirs
+        - toml >=0.9.4
+        - typed-ast >=1.4.0
+        - regex
+        - pathspec >=0.6,<1
+    test:
+      commands:
+        - black --help
+
+outputs:
```
Comment (xhochy) [inline recipe/meta.yaml:42]: The outputs is duplicated, probably eating up the `black` output.
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,33 +10,48 @@ source:
   sha256: c2edb73a08e9e0e6f65a0e6af18b059b8b1cdd5bef997d7a0b181df93dc81539
 
 build:
-  number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  number: 1
   skip: True  # [py<36]
```
Comment (xhochy) [inline recipe/meta.yaml:14]: If this can be `noarch`, we should get rid of the skip.
```suggestion
```

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,33 +10,47 @@ source:
...
-test:
-  commands:
-    - black --help
+outputs:
+  - name: {{ name }}
+    build:
+      noarch: python
+      script: "python -m pip install . -vv"
+      entry_points:
+        - black = black:patched_main
```
Comment (xhochy) [inline recipe/meta.yaml:21]: I would expect `blackd` to also be available here as it would be with the `black` python package.

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Pull request merged**